src/repository/SchedulesRepository.ts

```diff
@@ -0,0 +1,102 @@
+import { DateTime } from 'luxon';
+import { MongoClient } from 'mongodb';
+
+import { ScheduleEntity } from './ScheduleEntity';
+import { Repository } from './Repository';
+import { Logger } from '../logging/Logger';
+import { MomoErrorType } from '../logging/error/MomoErrorType';
+
+export const SCHEDULES_COLLECTION_NAME = 'schedules';
+
+export class SchedulesRepository extends Repository<ScheduleEntity> {
+  private logger: Logger | undefined;
+
+  constructor(
+    mongoClient: MongoClient,
+    private readonly deadScheduleThreshold: number,
+    private readonly scheduleId: string,
+    collectionPrefix?: string
+  ) {
+    super(mongoClient, SCHEDULES_COLLECTION_NAME, collectionPrefix);
+  }
+
+  setLogger(logger: Logger): void {
+    this.logger = logger;
+  }
+
+  async isActiveSchedule(): Promise<boolean> {
```

Comment:
Do we want to add a comment explaining a bit closer what this method actually does? 😅 

Even though I was there when it was written it still took me a second.